retrieval_service/app/app_test.py

```diff
@@ -12,97 +12,329 @@
+                    "2023-01-01 05:57:00", "%Y-%m-%d %H:%M:%S"
+                ),
+                arrival_time=datetime.strptime(
+                    "2023-01-01 12:13:00", "%Y-%m-%d %H:%M:%S"
+                ),
+                departure_gate="BAZ",
+                arrival_gate="QUX",
+            )
+        ]
+        return mock_flights
+
+    async def search_flights_by_airports(
+        self,
+        date,
+        departure_airport: Optional[str] = None,
+        arrival_airport: Optional[str] = None,
+    ) -> list[models.Flight]:
+        mock_flights = [
+            models.Flight(
+                id=1,
+                airline="search_flights_by_airports",
+                flight_number="FOOBAR",
+                departure_airport="FOO",
+                arrival_airport="BAR",
+                departure_time=datetime.strptime(
+                    "2023-01-01 05:57:00", "%Y-%m-%d %H:%M:%S"
+                ),
+                arrival_time=datetime.strptime(
+                    "2023-01-01 12:13:00", "%Y-%m-%d %H:%M:%S"
+                ),
+                departure_gate="BAZ",
+                arrival_gate="QUX",
+            )
+        ]
+        return mock_flights
+
+    async def close(self):
+        return
 
 
 @pytest.fixture(scope="module")
 def app():
-    cfg = AppConfig(
-        datastore=postgres.Config(
-            kind="postgres",
-            user=DB_USER,
-            password=DB_PASS,
-            database=DB_NAME,
-            host=IPv4Address(DB_HOST),
-        )
-    )
-    app = init_app(cfg)
+    mock_app_config = MagicMock()
+    app = init_app(mock_app_config)
     if app is None:
         raise TypeError("app did not initialize")
     return app
 
 
+@patch.object(datastore, "create", AsyncMock(return_value=MockDatastore()))
 def test_hello_world(app):
     with TestClient(app) as client:
         response = client.get("/")
         assert response.status_code == 200
         assert response.json() == {"message": "Hello World"}
 
 
-@pytest.mark.parametrize(
-    "params",
-    [
-        pytest.param(
-            {
-                "id": 1,
-            },
-            id="id_only",
-        ),
-        pytest.param({"iata": "sfo"}, id="iata_only"),
-    ],
-)
-def test_get_airport(app, params):
+get_airport_params = [
+    pytest.param(
+        {
+            "id": 1,
+        },
+        {
+            "id": 1,
+            "iata": "FOO",
+            "name": "get_airport_by_id",
+            "city": "BAR",
+            "country": "FOO BAR",
+        },
+        id="id_only",
+    ),
+    pytest.param(
+        {"iata": "sfo"},
+        {
+            "id": 1,
+            "iata": "FOO",
+            "name": "get_airport_by_iata",
+            "city": "BAR",
+            "country": "FOO BAR",
+        },
+        id="iata_only",
+    ),
+]
+
+
+@pytest.mark.parametrize("params, expected", get_airport_params)
+@patch.object(datastore, "create", AsyncMock(return_value=MockDatastore()))
```

Comment:
If it's not too much trouble, I think it's probably cleaner to move from using a single Mock to just using MagicMock and declaring what you want the return values to be using the mock directly in the test. It ties the return value closer to the test -- not only is this easier to update, but means you can have two different tests rely on two different mock values.